pkg/cmd/util/clientcmd/factory.go

```diff
@@ -774,3 +776,46 @@ func (c *clientCache) ClientForVersion(version *unversioned.GroupVersion) (*clie
 	c.clients[config.GroupVersion.String()] = client
 	return client, nil
 }
+
+// FindAllCanonicalResources returns all resource names that map directly to their kind (Kind -> Resource -> Kind)
+// and are not subresources. This is the closest mapping possible from the client side to resources that can be
+// listed and updated. Note that this may return some virtual resources (like imagestreamtags) that can be otherwise
+// represented. It collapses resources in different groups with the same name into a single value.
+// TODO: add a field to APIResources for "virtual" (or that points to the canonical resource).
+// TODO: fallback to the scheme when discovery is not possible.
+func FindAllCanonicalResources(d discovery.Discovery, m meta.RESTMapper) ([]unversioned.GroupResource, error) {
+	set := make(map[string]unversioned.GroupResource)
```

Comment:
I don't think they should (it's not the point of this method).  The
heuristic is resources with the same name share storage, and are henceforth
only migrated once.

On Jun 17, 2016, at 8:12 AM, David Eads <notifications@github.com> wrote:

In pkg/cmd/util/clientcmd/factory.go
<https://github.com/openshift/origin/pull/9380#discussion_r67500623>:

> @@ -774,3 +776,46 @@ func (c *clientCache) ClientForVersion(version *unversioned.GroupVersion) (*clie
>  	c.clients[config.GroupVersion.String()] = client
>  	return client, nil
>  }
> +
> +// FindAllCanonicalResources returns all resource names that map directly to their kind (Kind -> Resource -> Kind)
> +// and are not subresources. This is the closest mapping possible from the client side to resources that can be
> +// listed and updated. Note that this may return some virtual resources (like imagestreamtags) that can be otherwise
> +// represented. It collapses resources in different groups with the same name into a single value.
> +// TODO: add a field to APIResources for "virtual" (or that points to the canonical resource).
> +// TODO: fallback to the scheme when discovery is not possible.
> +func FindAllCanonicalResources(d discovery.Discovery, m meta.RESTMapper) ([]unversioned.GroupResource, error) {
> +	set := make(map[string]unversioned.GroupResource)

So if I'm reading this correctly, you're using this in a way that will fail
if I get a foo.group1 and foo.group2. Those would have a single entry in
the map, but they should have two.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/9380/files/fdc37b55eefd4f22bd8368bec25c51eb95c92009#r67500623>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p1WyVDmMM0Qd5MztOYGEpRTGIPM_ks5qMo80gaJpZM4I3dIQ>
.